server/boot/challenge.js

```diff
@@ -596,6 +605,107 @@ module.exports = function(app) {
       .subscribe(() => {}, next);
   }
 
+  function githubExport(req, res, next) {
+    const {
+        name,
+        solution
+    } = req.body;
+
+    function createRepo(token, repoName, description, callback) {
+        var authToken = 'token ' + token;
+
+        // NOTE ESLint gives a warning at the json 'auto_init' for not having
+        // CamelCase, but that is the format github wants it in
+        request({
+            url: 'https://api.github.com/user/repos',
+            method: 'POST',
+            json: {
+                name: repoName,
+                description: description,
+                auto_init: true
+            },
+            headers: {
+                Authorization: authToken,
+                'user-agent': 'FreeCodeCamp-Export'
+            }
+        }, function(error, response, body) {
+            callback(error, response, body);
+        });
+    }
+
+    function commitFile(token,
+        repoName,
+        solution,
+        challengeName,
+        dateComplete,
+        userName,
+        callback) {
+        // Setup Variables
+        var authToken = 'token ' + token;
+        var solutionEncoded = new Buffer('```\n' + solution + '\n```')
+            .toString('base64');
+        var fileName = challengeName + ' ' + dateComplete + '.md';
+        var commitMessage = 'Add Solution For ' + challengeName;
+
+        var json = {
+            message: commitMessage,
+            content: solutionEncoded
+        };
+        var url = 'https://api.github.com/repos/' + userName + '/' +
+            repoName + '/contents/' + fileName;
+
+        console.log(json + ' ' + url);
```

Comment:
As `json` here is actually a JavaScript object it will be converted to `"[object Object]"` here, so you need to `stringify` it first.
**A tip:** `console.log(json, url);` is slightly better then `console.log(json + ' ' + url);`